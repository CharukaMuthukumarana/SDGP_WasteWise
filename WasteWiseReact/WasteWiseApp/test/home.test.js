import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import Home from '../app/home';

// Mock the expo-router module
jest.mock('expo-router', () => ({
  router: {
    push: jest.fn(),
  },
}));

describe('<Home />', () => {
  it('renders correctly', () => {
    const { getByText } = render(<Home />);
    expect(getByText('Waste Wise')).toBeTruthy();
  });

  it('navigates to company login screen on company button press', () => {
    const { getByText } = render(<Home />);
    fireEvent.press(getByText('Company'));
    expect(require('expo-router').router.push).toHaveBeenCalledWith('/logins/companylogin');
  });

  it('navigates to recycling login screen on recycling button press', () => {
    const { getByText } = render(<Home />);
    fireEvent.press(getByText('Recycling'));
    expect(require('expo-router').router.push).toHaveBeenCalledWith('/logins/recyclinglogin');
  });

  it('navigates to driver login screen on driver button press', () => {
    const { getByText } = render(<Home />);
    fireEvent.press(getByText('Driver'));
    expect(require('expo-router').router.push).toHaveBeenCalledWith('/logins/driverlogin');
  });

  it('navigates to add trash cans screen on "Add Trash Cans" link press', () => {
    const { getByText } = render(<Home />);
    fireEvent.press(getByText('Add Trash Cans'));
    expect(require('expo-router').router.push).toHaveBeenCalledWith('/addtrashcan');
  });

  it('navigates to device data screen on "device data" link press', () => {
    const { getByText } = render(<Home />);
    fireEvent.press(getByText('device data'));
    expect(require('expo-router').router.push).toHaveBeenCalledWith('/devicedata');
  });
});
