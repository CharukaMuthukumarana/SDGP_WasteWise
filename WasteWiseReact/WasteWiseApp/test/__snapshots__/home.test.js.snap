// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<home /> renders correctly 1`] = `<home />`;
